npc/selector.go

```diff
@@ -61,8 +75,15 @@ type selector struct {
 	spec *selectorSpec
 }
 
-func (s *selector) matches(labelMap map[string]string) bool {
-	return s.spec.selector.Matches(labels.Set(labelMap))
+func (s *selector) matches(podsLabelMap *map[string]string, namespaceLabelMap *map[string]string) bool {
```

Comment:
The use of pointer-to-map here is a little unusual.
I think I can see what you are doing, and I can't say it's wrong, but I think it at least deserves a comment saying what the two parameters mean.

In many cases the code would be cleaner with three different `matches()` methods.

Reply:
added separate matches to make it cleaner